features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -149,28 +149,28 @@ internal class DatadogRumMonitor(
     override fun startView(key: Any, name: String, attributes: Map<String, Any?>) {
         val eventTime = getEventTime(attributes)
         handleEvent(
-            RumRawEvent.StartView(key, name, attributes, eventTime)
+            RumRawEvent.StartView(key, name, attributes.toMap(), eventTime)
         )
     }
 
     override fun stopView(key: Any, attributes: Map<String, Any?>) {
         val eventTime = getEventTime(attributes)
         handleEvent(
-            RumRawEvent.StopView(key, attributes, eventTime)
+            RumRawEvent.StopView(key, attributes.toMap(), eventTime)
         )
     }
 
     override fun addAction(type: RumActionType, name: String, attributes: Map<String, Any?>) {
         val eventTime = getEventTime(attributes)
         handleEvent(
-            RumRawEvent.StartAction(type, name, false, attributes, eventTime)
+            RumRawEvent.StartAction(type, name, false, attributes.toMap(), eventTime)
         )
     }
 
     override fun startAction(type: RumActionType, name: String, attributes: Map<String, Any?>) {
         val eventTime = getEventTime(attributes)
         handleEvent(
-            RumRawEvent.StartAction(type, name, true, attributes, eventTime)
+            RumRawEvent.StartAction(type, name, true, attributes.toMap(), eventTime)
```

Comment:
Why are these `toMap` calls needed now?

Reply:
This is to avoid having `ConcurrentModificationException` when we iterate over them later on another thread and they can be modified on the user-thread in the meantime. Quote from the PR description:

> We will make a copy of user-provided attributes in RumMonitor at the call site. This is to prevent ConcurrentModificationException which sometimes happen when we iterate over them later. Making a copy at the call-site has a benefit of doing it on the user-thread, which lowers the chance that concurrent modification is done at the same time, and if there is such case, exception will be thrown on the user thread and will be more visible.